src/dischat/storage/migrations/0007_pairing_atomicity.sql

```
@@ -0,0 +1,25 @@
+-- Preserve only the newest active session per Matrix user before enforcing
+-- the invariant for concurrent /pair requests.
+WITH ranked AS (
+    SELECT id, row_number() OVER (PARTITION BY mxid ORDER BY created_at DESC, id DESC) AS rn
+    FROM pairing_sessions
+    WHERE consumed_at IS NULL
+)
+UPDATE pairing_sessions
+SET consumed_at = NOW()
+WHERE id IN (SELECT id FROM ranked WHERE rn > 1);
```

SQLFluff (4.3.0)

[error] The 'WHERE' keyword should always start a new line.

(LT14)